homework_multithreading_3/src/main/java/org/application/Bar.java

```diff
@@ -0,0 +1,36 @@
+package org.application;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+
+import java.util.concurrent.*;
+
+public class Bar {
+    private final ExecutorService clientExecutor;
+    private final ExecutorService bartenderExecutor;
+    ThreadFactory bartenderFactory = new ThreadFactoryBuilder().setNameFormat("Bartender-%d").build();
+    ThreadFactory clientFactory = new ThreadFactoryBuilder().setNameFormat("Client-%d").build();
+
+    public Bar(int bartender, int clients) {
+        this.clientExecutor = Executors.newFixedThreadPool(clients, clientFactory);
+        this.bartenderExecutor = Executors.newFixedThreadPool(bartender, bartenderFactory);
+    }
+
+    public Future<String> acceptOrder() {
+        return clientExecutor.submit(new Client());
+    }
+
+    public Future<String> preparationDrinks(String drink) {
+        Bartender bartender = new Bartender(drink);
```

Comment:
Here seems you have infinity bartenders. You will always create new bartender by new order